CMakeLists.txt

```diff
@@ -32,32 +32,58 @@ endif()
 set(CMAKE_INCLUDE_CURRENT_DIR ON)
 set(CMAKE_INCLUDE_CURRENT_DIR_IN_INTERFACE ON)
 
-add_definitions(-DHAVE_ZLIB)
+include(FetchContent)
+include(GNUInstallDirs)
+set(INSTALL_CONFIGDIR ${CMAKE_INSTALL_LIBDIR}/cmake/${NIFTI_NIFTILIB_NAME})
+
 
 # add option to build shared libraries. The default is OFF to maintain the
 # current build behavior
+set( 1)
+  set( 1)
+
+option( NIFTI_INSTALL_NO_DEVELOPMENT "ITK build development flag." OFF)
+option( NIFTI_INSTALL_NO_LIBRARIES "ITK build development flag." OFF)
 option( BUILD_SHARED_LIBS "Toggle building shared libraries." OFF)
+if(BUILD_SHARED_LIBS)
+  set(NIFTI_LIBRARY_PROPERTIES
+    BUILD_SHARED_LIBS TRUE
+    POSITION_INDEPENDENT_CODE TRUE)
+  endif()
+
+# Generate a mapping from targets that need to be built to the sub-directories
+# in which they are located.
+macro(INSERT_INTO_DIRMAP _KEY _VALUE)
+  set("modules_${_KEY}" "${_VALUE}")
+endmacro()
+INSERT_INTO_DIRMAP(utils utils)
+INSERT_INTO_DIRMAP(znz znzlib)
+INSERT_INTO_DIRMAP(nifti2io nifti2)
+INSERT_INTO_DIRMAP(niftiio niftilib)
+INSERT_INTO_DIRMAP(cifti cifti)
+INSERT_INTO_DIRMAP(nifticdf nifticdf)
+
+# Fetch and populate the content of each sub-directory to provide access to
+# targets during cmake configure time
+foreach(module nifticdf niftiio nifti2io cifti)
+  FetchContent_Declare(
+  ${module}
+  SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/${modules_${module}})
+  FetchContent_GetProperties(${module})
+if(NOT ${${module}_POPULATED})
+  FetchContent_Populate(${module})
+  add_subdirectory(${${module}_SOURCE_DIR} ${${module}_BINARY_DIR})
+endif()
+endforeach()
+
 
-# Michael Hanke 2007-04-25 Note:
-# 'Utilities' is merged into the include install path to be able to easily
-# install into any other path if necessary. For the same reason lib, bin and
-# include dir for standalone builds are only set if not yet defined.
 
-###  USE AS PART OF ITK
+##  USE AS PART OF ITK
 if("${ITK_VERSION_MAJOR}" GREATER 0.0)
 #   message(STATUS "---------------------ITK ZLIB  ${ITK_ZLIB_LIBRARIES}")
   add_definitions(-DITKZLIB)
   set(PACKAGE_PREFIX "ITK")
   set(NIFTI_LIBRARY_PROPERTIES ${ITK_LIBRARY_PROPERTIES})
-  set(NIFTI_INSTALL_BIN_DIR "${ITK_INSTALL_BIN_DIR_CM24}")
-  set(NIFTI_INSTALL_LIB_DIR "${ITK_INSTALL_LIB_DIR_CM24}")
-
-  if(ITK_USE_FLAT_DIRECTORY_INSTALL)
-    set(NIFTI_INSTALL_INCLUDE_DIR "${ITK_INSTALL_INCLUDE_DIR_CM24}")
```

Comment:
Why is this removed?

Reply:
I have not fully figured out the installation yet. The various binaries and libraries happily install but the linking is broken. I have a separate build working that linked fine before and after installation. I was going to work through what I did differently tomorrow. Once I have that fixed I can happily restored the installation functionality for ITK/non-ITK installations once I have this working

Reply:
This whole section should be protected by the 

if("${ITK_VERSION_MAJOR}" GREATER 0.0)  conditional and should not affect your normal builds.